src/main/host/descriptor/socket/netlink.rs

```diff
@@ -0,0 +1,1215 @@
+
+        let mut buffer = Cursor::new(Vec::new());
+        // Send the interface addresses
+        for interface in &common.interfaces {
+            let address = interface.address.octets();
+            let broadcast = if interface.prefix_len == 32 {
+                address
+            } else {
+                Ipv4Addr::from(
+                    (0xffff_ffff_u32) >> u32::from(interface.prefix_len)
+                        | u32::from(interface.address),
+                )
+                .octets()
+            };
+            let mut label = Vec::from(interface.label.as_bytes());
+            label.push(0); // Null-terminate
+
+            // List of attribtes sent with the response for the current interface
+            let attrs = [
+                // I don't know the difference between IFA_ADDRESS and IFA_LOCAL. However, Linux
+                // provides the same address for both attributes, so I do the same.
+                // Run `strace ip addr` to see.
+                Rtattr::new(None, Ifa::Address, Buffer::from(&address[..])).unwrap(),
+                Rtattr::new(None, Ifa::Local, Buffer::from(&address[..])).unwrap(),
+                Rtattr::new(None, Ifa::Broadcast, Buffer::from(&broadcast[..])).unwrap(),
+                Rtattr::new(None, Ifa::Label, Buffer::from(label)).unwrap(),
+            ];
+            let ifaddrmsg = Ifaddrmsg {
+                ifa_family: RtAddrFamily::Inet,
+                ifa_prefixlen: interface.prefix_len,
+                // IFA_F_PERMANENT is used to indicate that the address is permanent
+                ifa_flags: IfaFFlags::new(&[IfaF::Permanent]),
+                ifa_scope: libc::c_uchar::from(interface.scope),
+                ifa_index: interface.index,
+                rtattrs: RtBuffer::from_iter(attrs),
+            };
+            let nlmsg = {
+                let len = None;
+                let nl_type = Rtm::Newaddr;
+                // The NLM_F_MULTI flag is used to indicate that we will send multiple messages
+                let flags = NlmFFlags::new(&[NlmF::Multi]);
+                // Use the same sequence number as the request
+                let seq = Some(nlmsg.nl_seq);
+                let pid = None;
+                let payload = NlPayload::Payload(ifaddrmsg);
+                Nlmsghdr::new(len, nl_type, flags, seq, pid, payload)
+            };
+            nlmsg.to_bytes(&mut buffer).unwrap();
+        }
+        // After sending the messages with the NLM_F_MULTI flag set, we need to send the NLMSG_DONE message
+        let done_msg = {
+            let len = None;
+            let nl_type = Nlmsg::Done;
+            let flags = NlmFFlags::new(&[NlmF::Multi]);
+            // Use the same sequence number as the request
+            let seq = Some(nlmsg.nl_seq);
+            let pid = None;
+            // Linux also emits 4 bytes of zeroes after the header. See `strace ip addr`
+            let payload: NlPayload<Nlmsg, u32> = NlPayload::Payload(0);
```

Comment:
This is strange. I wonder if there is an extended header and we're missing the suffix part of it or something like that? If you could find out what the linux kernel source code is doing, it would be helpful to have a link to that here as a comment.

Reply:
Yeah I was surprised by this as well. Yeah it's a good idea to dig into the linux source code.